cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -163,6 +163,10 @@ func NewStaticAutoscaler(
 	}
 
 	clusterStateRegistry := clusterstate.NewClusterStateRegistry(autoscalingContext.CloudProvider, clusterStateConfig, autoscalingContext.LogRecorder, backoff)
+	autoscalingContext.ScaleDownCandidatesObservers = []scaledown.ScaleDownCandidatesObserver{clusterStateRegistry}
+	if p, ok := processors.ScaleDownNodeProcessor.(scaledown.ScaleDownCandidatesObserver); ok {
```

Comment:
This feels a bit hacky. In particular, while in principle we allow multiple observers, in practice they are hardcoded here, making introduction of new observers tricky. How about passing a slice of observers to register in autoscaling opts? Then the code here would register `clusterStateRegistry` and all the extra observers passed through the options.

Reply:
Moving ScaleDownCandidatesObservers to the autoscaling opts creates an import cycle. 

We already have `if opts.ParallelDrain` condition. Maybe I can make processor assignment there and add an observer there.

Reply:
Hm... if cyclic imports are the problem, maybe bundling the observer interface with other scale down interfaces is not the right place. WDYT about moving all the newly added code to a dedicated package?

Reply:
It seems for me that autoscaling opts is a set of options set via flags +-, so autoscaling context object sounds like a better place for scale down candidates observers. Also, we are now creating the initial list there, instead of static autoscaler, so maybe it solves initial issue?

Reply:
My concern here is: how can I register a new scaledown candidates observer without modifying code of StaticAutoscaler? Or to put it differently: how can we make static autoscaler unaware of who wants to observe scale down candidates?

Reply:
Hm, I think the common approach that we used so far is to modify processors. I introduced ScaleDownCandidatesObservers there, please take a look